### ISSUE TITLE
[Enhancement] autoScale add ability to enable auto scaling per axis edge

### ISSUE THREAD

**Issue by exlted** [flot/flot#1604]
Prior to the recent updates, I was able to set just a min or just a max value which would lock that edge of the axis, allowing the other to scale depending on my data. In the most recent update I've been having trouble getting my axes to act consistently in this situation.

For example, I have a bar graph that we never want to show negative bars on, but our data can be negative. In this case, the prior version would hide that singular bar as it was out of range, in this version I seem to have the options of setting both min and max, or showing all my data. 

This issue manifested as flot not utilizing the min or max values I provided initially, then utilizing them once I caused the graphs to redraw.

**Comment by andrewdove1** [comment]
If you are not autoscaling then you can set auto scale to none and set min and max.
Otherwise there is a hook for custom autoscale behavior called setRange.

**Comment by exlted** [comment]
So you would consider the ability to enforce a minimum and have a floating maximum to be custom enough to require a setRange hook? 

If that's the assumption, then that works for me

**Comment by andrewdove1** [comment]
Yes that's the way it works now

**Comment by atmgrifter00** [comment]
He @exlted, I'd be curious to know if the given suggestion ultimately solved your problem, and we can consider this issue closed.

**Comment by exlted** [comment]
Due to other issues with the recent changes we currently haven't updated the version of flot we're using (And knowing my company, we likely won't for quite a while... *sigh*). 

Since we're able to work around the issue without needing to make changes to flot itself, I don't see a necessity to put time into implementing this enhancement into the base library.

**Comment by atmgrifter00** [comment]
I'd love to hear what the other issues are that are getting in your way. Hopefully, they are the ones listed in the upgrade notes. If so, perhaps a good resolution to the '[Backward Compatibility](https://github.com/flot/flot/issues/1612)' issue that was raised would allow you get on the new version.

**Comment by exlted** [comment]
The primary issue that was blocking us was the the bar graph hovering: #1603.  I logged everything major I ran into last week when I was attempting to handle the update, but I don't remember any of the other minor issues that I ran into while testing, sadly. 

We probably won't end up attempting another update at least until after #1602 gets cleared up since we're currently running of a modified older version that include a version of my changes.

**Comment by jarekdobrzanski** [comment]
I haven't updated yet as I have a couple of corehacks and trying to figure out what to do with them. Without checking I'm sure this problem will break my code as I use autoscaling with single edge fixated a lot.

**Comment by atmgrifter00** [comment]
Yeah, I think we need to investigate how to re-introduce this functionality (autoscaling with single edge fixated). I believe I understand the reason why it was broken within the engineering-flot fork, but we (my company) should probably spend some time figuring out how to service this need in addition to the one that caused us to break it in the first place.

**Comment by skortchmark9** [comment]
I'm running into this problem as well. I have a time series plot that I want to scroll with my data, but I want users to be able to manually set yaxis bounds. I think flot's built-in autoscaling is quite good, so I'd prefer to use it vs. rolling my own.

I've attached the workaround I'm using here - perhaps this will help someone fix the problem upstream:
```javascript
// copy these two functions from jquery.flot.js
function widenMinMax(minimum, maximum) {
    ...
}

function autoScaleAxis(axis) {
    ...
}

// in flot options
hooks: {
    setRange(plot, axis, autoScale) {
        autoScaleAxis(axis);
        axis.min = axis.autoScaledMin;
        axis.max = axis.autoScaledMax;
    },
}
```

This seems to be working for me.

**Comment by mikh3x4** [comment]
I'm not sure if my issue is identical but I want the lower axis range to always be 0 and I want the top to be picked automatically based on data. 

A hacky solution I came up with is adding an extra invisible data-series with a point at 0 to ensure to is always shown. Hope it help someone if they randomly stumble on this post like I have
